ghc-lib-gen.cabal

```diff
@@ -12,16 +13,36 @@ copyright:           Digital Asset 2018-2019
 category:            Development
 build-type:          Simple
 extra-source-files:  ChangeLog.md
-cabal-version:       >=1.10
+
+library
+  build-depends:       base >=4.10
+                     , process >=1.6
+                     , filepath >=1.4
+                     , containers
+                     , directory
+                     , optparse-applicative
+                     , extra >=1.6
+  hs-source-dirs:    ghc-lib-gen/src
+  exposed-modules:   Ghclibgen
+                     GhclibgenOpts
+                     Paths_ghc_lib_gen
+  autogen-modules:   Paths_ghc_lib_gen
+  default-language:  Haskell2010
+  ghc-options:       -Wmissing-signatures
 
 executable ghc-lib-gen
-  main-is:             Main.hs
+  main-is:           Main.hs
   build-depends:       base >=4.10
                      , process >=1.6
                      , filepath >=1.4
                      , containers
                      , directory
+                     , optparse-applicative
                      , extra >=1.6
-  hs-source-dirs:      ghc-lib-gen/src
-  default-language:    Haskell2010
-  ghc-options: -Wmissing-signatures
+                     , ghc-lib-gen
+  other-modules:     Ghclibgen
+                     GhclibgenOpts
+                     Paths_ghc_lib_gen
```

Comment:
Those modules shouldn’t be listed in `other-modules` as you are getting them from the library. You probably have to remove the `hs-source-dirs` line and change `main-is` to `src/Main.hs` and/or use different `hs-source-dirs` for the executable and the library.

Reply:
^ Ditto but your advise on `ghc-lib-gen/src/Main.hs` appears to be the missing ingredient. Thanks!